### PR TITLE
feat: handcuffing and dependency checks

### PR DESCRIPTION
**Pull request by mafewtm**
## Description

- Adds handcuffing/unhandcuffing with handcuffs and handcuff keys
- Adds dependency and version checking
- Fixes callsign command job checking
- Fixes the adding of vehicle mods

## Checklist

- [ ] I have personally loaded this code into an updated Qbox project and checked all of its functionality.
- [x] My pull request fits the contribution guidelines & code conventions.

### CODE REVIEW

File: server/interactions.lua [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    local playerCoords = GetEntityCoords(playerPed)
+    local targetId, _, _ = lib.getClosestPlayer(playerCoords, 1.0)
+
+    if not targetId or Player(targetId).state.handcuffed then return end
+
+    TriggerClientEvent('qbx_police:client:handcuffPlayer', source)
+
+    local targetCuffed = lib.callback.await('qbx_police:client:getHandcuffed', targetId)
+
+    if not targetCuffed then return end
```
Comment (Manason) [inline server/interactions.lua:13]: These seem like they could be grouped (no whitespace needed?)

File: server/interactions.lua [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+    exports.ox_inventory:RemoveItem(source, 'handcuffs', 1)
```
Comment (Manason) [inline server/interactions.lua:15]: config option to determine whether handcuffs are consumed?

**Review by Manason — APPROVED** [review]

File: server/interactions.lua (hunk 2 above)
Comment (mafewtm) [reply server/interactions.lua:15]: Why does there need to be a config option? It seems pretty obvious that we'd want the handcuffs removed upon completion of the animations and failure of the skillcheck.

Comment (Manason) [reply server/interactions.lua:15]: I suspect many servers will not want to consume handcuffs.

File: client/interactions.lua [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+    local success = lib.skillCheck('medium')
+
+    LocalPlayer.state.invBusy = true
+
+    lib.playAnim('mp_arrest_paired', 'crook_p2_back_right', 8.0, 8.0, 3750, 48, 0.0, false, 0, false)
+    Wait(3750)
+
+    while lib.skillCheckActive() do
+        Wait(0)
+    end
```
[the remark this answers is not in the document]
Comment (Manason) [reply client/interactions.lua:46]: I'm a bit confused here. How will the skill check be active when we get to this code?

[the remark this answers is not in the document]
Comment (mafewtm) [reply client/interactions.lua:46]: With how it's currently set up, it wont be. But servers may want to edit this so if they decide to make it longer, we want the code to wait before we get to any `if success then` code

File: server/interactions.lua (hunk 2 above)
Comment (Manason) [reply server/interactions.lua:15]: Most servers currently use handcuffs as an item that isn't consumed and don't use a separate key system. Alternative ideas are good, but we should continue to support the most popular one as that seems to be what people want.

**Pull request merged**